python/lsst/meas/algorithms/maskStreaks.py

```diff
@@ -0,0 +1,788 @@
+        X = np.array([x, y]).T
+        nClusters = 1
+
+        # Put line parameters in clusters by starting with all in one, then
+        # subdividing until the parameters of each cluster have std dev=1.
+        # If nClusters == len(lines), each line will have its own 'cluster', so
+        # the standard deviations of each cluster must be zero and the loop
+        # is guaranteed to stop.
+        while True:
+            kmeans = KMeans(n_clusters=nClusters, n_init='auto').fit(X)
+            clusterStandardDeviations = np.zeros((nClusters, 2))
+            for c in range(nClusters):
+                inCluster = X[kmeans.labels_ == c]
+                clusterStandardDeviations[c] = np.std(inCluster, axis=0)
+            # Are the rhos and thetas in each cluster all below the threshold?
+            if (clusterStandardDeviations <= 1).all():
+                break
+            nClusters += 1
+
+        # The cluster centers are final line estimates
+        finalClusters = kmeans.cluster_centers_.T
+
+        # Rescale variables:
+        finalRhos = finalClusters[0] * self.config.rhoBinSize
+        finalThetas = finalClusters[1] * self.config.thetaBinSize
+        result = LineCollection(finalRhos, finalThetas)
+        self.log.info("Lines were grouped into %s potential streak(s)", len(finalRhos))
+
+        return result
+
+    def _fitProfile(self, lines, maskedImage):
+        """Fit the profile of the streak.
+
+        Given the initial parameters of detected lines, fit a model for the
+        streak to the original (non-binary image). The assumed model is a
+        straight line with a Moffat profile.
+
+        Parameters
+        ----------
+        lines : `LineCollection`
+            Collection of guesses for `Line`s detected in the image.
+        maskedImage : `lsst.afw.image.maskedImage`
+            Original image to be used to fit profile of streak.
+
+        Returns
+        -------
+        lineFits : `LineCollection`
+            Collection of `Line` profiles fit to the data.
+        finalMask : `np.ndarray`
+            2d mask array with detected streaks=1.
+        """
+        data = maskedImage.image.array
+        weights = maskedImage.variance.array**-1
+        # Mask out any pixels with non-finite weights
+        weights[~np.isfinite(weights) | ~np.isfinite(data)] = 0
+
+        lineFits = LineCollection([], [])
+        finalLineMasks = [np.zeros(data.shape, dtype=bool)]
+        nFinalLines = 0
+        for line in lines:
+            line.sigma = self.config.invSigma**-1
+            lineModel = LineProfile(data, weights, line=line)
+            # Skip any lines that do not cover any data (sometimes happens because of chip gaps)
+            if lineModel.lineMaskSize == 0:
+                continue
+
+            fit, chi2, fitFailure = lineModel.fit(dChi2Tol=self.config.dChi2Tolerance, log=self.log)
+            if fitFailure:
+                self.log.warning("Streak fit failed.")
```

Comment:
Please elaborate on this warning message, with at least a "on line X out of nLines" or other identifier. If this is expected to happen, consider downgrading this to an `info` message.

Reply:
We are allowed to log statistics on streaks as long as there is no way to determine individual streak characteristics, so perhaps simply count the number of streak fit failures and log a single warning message such as `"Streak fitting failed on XX lines"`. My concern is about being spammed with unhelpful warning messages in crowded fields.